### PR TITLE
Matrix infer shape

### PR DESCRIPTION
**Pull request by vetlewi**
The `Matrix` class can now infer the `shape` parameter at initialization if both `Ex` and `Eg` variables are set but shape is not. Also added the `fill` method to make it easy to add data to the histogram.

### CODE REVIEW

**Comment by fzeiser** [comment]
- Fill:
I see the point of the fill method, which corresponds to the method in root.

- Shape:
Can you explain the part on the shape, and why this is helpful/necessary? You don't need to provide a shape if you initialize Matrix with a non-zero values. In order to have the `Matrix` resemble a `np.array`, I would rather suggest to have following behavior:

- remove `shape` from the initialization (print Deprec warning)
- if values is not provided, return an error
- create a `ZeroMatrix`, that acts somewhat like `np.zeros()` and infers the shape from Eg and Ex.

What do you think?

**Comment by vetlewi** [comment]
> * Shape:
>   Can you explain the part on the shape, and why this is helpful/necessary? You don't need to provide a shape if you initialize Matrix with a non-zero values. In order to have the `Matrix` resemble a `np.array`, I would rather suggest to have following behavior:
>   
>   * remove `shape` from the initialization (print Deprec warning) [though I don't think anyone used the initialization with `shape`]
>   * if values is not provided, return an error
>   * create a `ZeroMatrix`, that acts somewhat like `np.zeros()` and infers the shape from Eg and Ex.
> 
> What do you think?

This PR was to make it easier to create a new matrix with zeros. Currently one would do that by
```python
Ex = np.linspace(0, 10, 101)
Eg = np.linspace(0, 10, 101)
mat = np.Matrix(shape=(len(Ex),len(Eg)), Ex=Ex, Eg=Eg) 
```
but the `shape` parameter is redundant as one could infer shape from `Ex` and `Eg`. Reason `shape` wasn't removed in the PR was to avoid breaking anything inside ompy that may or may not use it.

**Comment by fzeiser** [comment]
I double checked: OMpy doesn't use it, only potentially user code. But I didn't create any example with it either. Usually, when I wanted to have matrixes with zeros as entries i passed `values=np.zeros((len(Ex),len(Eg)))`. My thought now was just that it might be better to just get rid of `shape` at this point and make the code somewhat clearer. If you don't object, I'll change this PR accordingly.

**Comment by vetlewi** [comment]
> I double checked: OMpy doesn't use it, only potentially user code. But I didn't create any example with it either. Usually, when I wanted to have matrixes with zeros as entries i passed `values=np.zeros((len(Ex),len(Eg)))`. My thought now was just that it might be better to just get rid of `shape` at this point and make the code somewhat clearer. If you don't object, I'll change this PR accordingly.

Go right ahead!

File: ompy/matrix.py [hunk 1]
```diff
@@ -99,24 +101,29 @@ def __init__(self,
...
             state: An enum to keep track of what has been done to the matrix.
                 Can also be a str. like in ["raw", "unfolded", ...]
+            shape: Depreciated. Use `ZerosMatrix` instead.
 
         """
+        if shape is not None:
+            warnings.warn("Creating a Matrix with zeros as entries by the "
+                          "shape argument is depreciated. Use ZerosMatrix "
+                          "instead.", DeprecationWarning)
+            values = ZerosMatrix(shape=shape, Ex=Ex, Eg=Eg).values
```
Comment (fzeiser) [inline ompy/matrix.py:113]: this is slightly ugly, but would preserve the current functionality  for some time (until one cleans up)

**Review by fzeiser — APPROVED** [review]
I think I'm done with the changes. Any comments?

File: ompy/matrix.py (hunk 1 above)
Comment (vetlewi) [reply ompy/matrix.py:113]: I think it looks alright, gets the job done. Isn't meant to last forever anyways:)

File: ompy/matrix.py [hunk 2]
```diff
@@ -756,6 +763,17 @@ def diagonal_elements(self) -> Iterator[Tuple[int, int]]:
...
 
+    def fill(self, Eg: float, Ex: float, count: Optional[float] = 1) -> None:
+        """ Add counts to the bin containing Eg and Ex.
+        Args:
+            Eg: (float) x-axis value
+            Ex: (float) y-axis values
+            count: (Optional) number to add to the bin.
+        Returns: None
+        """
+        print(Eg, Eg, self.values)
```
Comment (vetlewi) [inline ompy/matrix.py:774]: Why do we print?

**Review by vetlewi — COMMENTED** [review]
Looks good. Only thing I'm wondering about is that `print` command.

File: ompy/matrix.py (hunk 2 above)
Comment (fzeiser) [reply ompy/matrix.py]: Removed it now. Stupid me debugging.

**Pull request merged**